build.gradle

```diff
@@ -137,6 +137,8 @@ allprojects {
         kotlinOptions.freeCompilerArgs += "-progressive"
         // Binary compatibility support
         kotlinOptions.freeCompilerArgs += ["-Xdump-declarations-to=${buildDir}/visibilities.json"]
+        // Remove null assertions to get smaller bytecode on Android
+        kotlinOptions.freeCompilerArgs += ["-Xno-param-assertions", "-Xno-receiver-assertions", "-Xno-call-assertions"]
```

Comment:
Does it finally work?

Reply:
I'm using those in prod and can confirm that they work and reduce dex size by a large factor.

But I've exchanged a lot with R8 team on a few things and I was told that in the future it may be better to not apply those as R8 does use those intrinsics for some of their optimisations.
As of R8 1.5.51 (Last one currently for future AS 3.5 release) those compile flags still do a better job than R8, but it may not be the case for future 1.6 and it should be discussed with R8 tmea I suppose.

Reply:
They don't really work here. It seems to be a bug in `kotlin-multiplatform` plugin.

Reply:
Fixed now.